components/tidb_query/src/codec/datum_codec.rs

```diff
@@ -194,12 +239,16 @@ pub trait DatumFlagAndPayloadEncoder: BufferWriter + DatumPayloadEncoder {
         Ok(())
     }
 
-    fn write_datum_duration(&mut self, val: Duration) -> Result<()> {
+    fn write_datum_duration_int(&mut self, val: Duration) -> Result<()> {
         self.write_u8(datum::DURATION_FLAG)?;
         self.write_datum_payload_i64(val.to_nanos())?;
         Ok(())
     }
 
+    fn write_datum_date_time_int(&mut self, val: DateTime, ctx: &mut EvalContext) -> Result<()> {
```

Comment:
The `write_datum_date_time_int` should be `write_datum_datetime_int` to keep consistent name convention (the former code copied from doesn't follow this convention, it's better to fix it here).